File: problemsets/Codeforces/Python/A1031.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
w,h,k=map(int,input().split());r=0
for _ in [0]*k:
  r+=(w+h<<1)-4;w-=4;h-=4
print(r)

File: problemsets/Codeforces/Java/A1031.java
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
import java.util.*;

public class A1031 {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int w = in.nextInt(), h = in.nextInt(), k = in.nextInt();
		int r = 0;
		for (int i = 1; i <= k; i++, w-=4, h-=4) r += 2*(h+w-2);
		System.out.println(r);
	}
}